### DISCUSSION TITLE
Bevy UI

### DISCUSSION THREAD

**Discussion by Barugon** [bevyengine/bevy#2857] [Q&A] [answered]
Has there been any movement on UI? Should I just forget about it and use egui? If I do then how difficult will it be to switch over to Bevy's UI when it actually exists?

**Comment by minecrawler** [comment, marked as answer]
If you need UI right now, there are [a few options available](https://bevyengine.org/assets/#ui). It seems, however, that the new UI architecture for Bevy will not land for quiet some time. There is a lot of movement, though, and you can dive into it:

- In the [v0.7 planning thread](https://github.com/bevyengine/bevy/discussions/2799#discussioncomment-1302474), people agree that UI is (one of) the next important thing, however it will probably need quiet some time to get right (will not land in 0.7, and probs also not in 0.8)
- There is a lot of discussion and many ideas in the [UI](https://github.com/bevyengine/rfcs/pull/1) [RFCs](https://github.com/bevyengine/rfcs/pull/22) and the [UI-issue](https://github.com/bevyengine/bevy/issues/254), but no consensus on how to really move forward. Probably pending a cart-hammer, but I guess he'll only take that on for 0.7, once the renderer rework is done.
- Not to mention that there are things all over the place which would be good to have for the new UI, like ray-casting, picking, [primitive shapes](https://github.com/bevyengine/rfcs/pull/12), [entity relations](https://github.com/bevyengine/rfcs/pull/18), [Parent/Child hierarchy improvements](https://github.com/bevyengine/bevy/discussions/2572)
